apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -1637,6 +1745,68 @@ def create_ingestion_dag(
+                        dimensions=dimensions,
+                        vector_type=vector_type,
+                        array_name=TRAINING_INPUT_VECTORS_ARRAY_NAME
+                    )
+                    training_source_type = "TILEDB_ARRAY"
+                    group.close()
+
+                    # Then sample training_sample_size vectors from input_vectors.
+                    # The number of complete batches.
+                    num_batches = math.ceil(in_size / input_vectors_batch_size)
+                    # The number of data points to sample from each batch.
+                    samples_per_batch = 0 if num_batches == 0 else math.floor(training_sample_size / num_batches)
+                    # Calculate the remaining data points that need to be sampled (i.e. b/c the numbers were not perfectly divisible).
+                    remaining_samples = training_sample_size - (num_batches * samples_per_batch)
+                    # Calculate how many of the remaining data points to sample from each batch.
+                    extra_samples_per_batch = remaining_samples if num_batches == 0 else math.ceil(remaining_samples / num_batches)
+                    # Calculate at which index to stop sampling an extra_samples_per_batch from.
+                    idx_to_stop_adding_extra_samples = 0 if extra_samples_per_batch == 0 else remaining_samples / extra_samples_per_batch
+
+                    idx = 0
+                    num_sampled = 0
+                    for i in range(0, in_size, input_vectors_batch_size):
+                        # What vectors to read from the source_uri.
+                        start = i
+                        end = i + input_vectors_batch_size
+                        if end > size:
+                            end = size
+
+                        # How many of those vectors to sample in this batch.
+                        num_for_batch = samples_per_batch
+                        if idx < idx_to_stop_adding_extra_samples:
+                            num_for_batch += extra_samples_per_batch
+
+                        submit(
+                            random_sample_from_input_vectors,
+                            source_uri=source_uri,
+                            source_type=source_type,
+                            vector_type=vector_type,
+                            dimensions=dimensions,
+                            vector_start_pos=start,
+                            vector_end_pos=end,
+                            batch=input_vectors_per_work_item,
+                            random_sample_size=num_for_batch,
+                            output_source_uri=training_source_uri,
+                            output_start_pos=num_sampled,
+                            config=config,
+                            verbose=verbose,
+                            name="read-random-sample-" + str(idx),
+                            resources={"cpu": str(threads), "memory": "12Gi"},
```

Comment:
Do we actually need "12Gi" memory for this task?

Reply:
Talked offline - lowered this to `1Gi` and updated so that we decrease the number of workers needed and how much memory we use for each - the end result is we will have less workers with less memory, which iterate over smaller batches of data more times.